rosplan_planning_system/src/PlanDispatch/EsterelPlanDispatcher.cpp

```diff
@@ -3,396 +3,510 @@
+                }
+                                
+                // check action edges
+                bool edges_activate_action = true;
+                std::vector<int>::iterator eit = node.edges_in.begin();
+                for (; eit != node.edges_in.end(); ++eit) {
+                    if(!edge_active[(*eit)]) {
+                        edges_activate_action = false;
+                        continue;
+                    }
+                }
+                if(!edges_activate_action) continue;
+
+                // handle completion of an action
+                if(node.node_type == rosplan_dispatch_msgs::EsterelPlanNode::ACTION_END && action_completed[node.action.action_id]) {
+
+                    ROS_INFO("KCL: (%s) %i: action %s completed",
+                            ros::this_node::getName().c_str(),
+                            node.action.action_id,
+                            node.action.name.c_str());
+
+                    finished_execution = false;
+                    state_changed = true;
+
+                    // deactivate incoming edges
+                    std::vector<int>::const_iterator ci = node.edges_in.begin();
+                    for(; ci != node.edges_in.end(); ci++) {
+                        edge_active[*ci] = false;
+                    }
+
+                    // activate new edges
+                    ci = node.edges_out.begin();
+                    for(; ci != node.edges_out.end(); ci++) {
+                        edge_active[*ci] = true;
+                    }
+                }
+
+                if(timeout_actions) {
+
+                    // check time bounds on edges
+                    bool times_activate_action = true;
+                    eit = node.edges_in.begin();
+                    for (; eit != node.edges_in.end(); ++eit) {
+
+                        rosplan_dispatch_msgs::EsterelPlanEdge edge = current_plan.edges[*eit];
+
+                        //define a minimum and maximum dispatch time for each edge
+                        float minimum_dispatch_time = node_real_dispatch_time[edge.source_ids[0]] + edge.duration_lower_bound - planStartTime;
+                        float maximum_dispatch_time = node_real_dispatch_time[edge.source_ids[0]] + edge.duration_upper_bound - planStartTime;
+
+                        // widen bounds based on parameter (default by 0)
+                        minimum_dispatch_time = minimum_dispatch_time - action_timeout_fraction*minimum_dispatch_time;
+                        maximum_dispatch_time = maximum_dispatch_time + action_timeout_fraction*maximum_dispatch_time;
+
+                        // check the current time with the lower bound
+                         double NOW = ros::Time::now().toSec();
+                         if (NOW - planStartTime < minimum_dispatch_time) { 
+                            if(bl_end && node.node_type == rosplan_dispatch_msgs::EsterelPlanNode::ACTION_END && action_completed[node.action.action_id]) {
```

Comment:
Remove lines 211 to 214:
  if(bl_end ...) { ... }
We will have to revisit this part later when there are proper parameters that the user can set to control exactly when the plan is cancelled.